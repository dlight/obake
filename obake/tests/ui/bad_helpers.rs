mod structs {
    #[obake::versioned]
    #[obake(version("0.1.0"))]
    #[obake(inherit)]
    struct Foo {}

    #[obake::versioned]
    #[obake(version("0.1.0"))]
    #[obake(cfg("0.1.0"))]
    struct Bar {}
}

mod enums {
    #[obake::versioned]
    #[obake(version("0.1.0"))]
    #[obake(inherit)]
    enum Foo {}

    #[obake::versioned]
    #[obake(version("0.1.0"))]
    #[obake(cfg("0.1.0"))]
    enum Bar {}

    #[obake::versioned]
    #[obake(version("0.1.0"))]
    enum Baz {
        #[obake(inherit)]
        Variant,
    }
}

fn main() {}
