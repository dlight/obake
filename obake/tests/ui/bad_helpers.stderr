error: `#[obake(inherit)]` not valid in this context
 --> $DIR/bad_helpers.rs:4:13
  |
4 |     #[obake(inherit)]
  |             ^^^^^^^

error: `#[obake(cfg(...))]` not valid in this context
 --> $DIR/bad_helpers.rs:9:17
  |
9 |     #[obake(cfg("0.1.0"))]
  |                 ^^^^^^^

error: `#[obake(inherit)]` not valid in this context
  --> $DIR/bad_helpers.rs:16:13
   |
16 |     #[obake(inherit)]
   |             ^^^^^^^

error: `#[obake(cfg(...))]` not valid in this context
  --> $DIR/bad_helpers.rs:21:17
   |
21 |     #[obake(cfg("0.1.0"))]
   |                 ^^^^^^^

error: `#[obake(inherit)]` not valid in this context
  --> $DIR/bad_helpers.rs:27:17
   |
27 |         #[obake(inherit)]
   |                 ^^^^^^^
